### DISCUSSION TITLE
Feature Request: Private notes on issues and PRs

### DISCUSSION THREAD

**Discussion by Vexed01** [community/community#60808] [Projects and Issues]
### Select Topic Area

Product Feedback

### Feature Area

Issues

### Body

For both issues and PRs, it would be helpful to include a private notes section so that users can note down why an issue or PR is relevant to them when they subscribe to it.

Use case: I subscribe to an issue or PR but then (maybe after a period of inactivity on the issue) forget why it is relevant to me. Adding a facility to privately note down the reason for subscribing would be helpful.

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by ggnall** [comment]
This is relevant to us since we work in public with a public repo. Sometimes we want to log sensitive data (e.g. what users requested something)

**Comment by ajor** [comment]
I'd also find this useful as a code reviewer.

On big PRs I like to make notes to remind myself about things as I jump around. I do it by leaving pending comments right now, but this method falls apart if I want to submit real comments as part of a review - I have to delete all my private pending comments and then re-add them if there are still things I want I want to remind myself about when I come back to the PR later.
